### ISSUE TITLE
Unable to Create Citations via YAML Frontmatter

### ISSUE THREAD

**Issue by TheCedarPrince** [gollum/gollum#1470]
Currently, I have a .md that I am trying to render a citation in via YAML front matter and the instructions you gave here: github.com/gollum/gollum/wiki/BibTeX-and-Citations

So far, I am having no luck with doing this as it seems to ignore that I have installed the ruby gems for rendering the citations.

Here is an example of my markdown file:

```mkd
---
bibliography: reference.bib
---

Hello world!

[@clearZettelkastenHowOne2020]

```

Here is my .bib file for reference:

```bib
@online{clearZettelkastenHowOne2020,
  title = {Zettelkasten — {{How One German Scholar Was So Freakishly Productive}}},
  author = {Clear, David B.},
  date = {2020-01-06T19:34:37.216Z},
  journaltitle = {Medium},
  url = {https://writingcooperative.com/zettelkasten-how-one-german-scholar-was-so-freakishly-productive-997e4e0ca125},
  urldate = {2020-03-08},
  abstract = {Luhmann wrote over 70 books and more than 400 scholarly articles using the Zettelkasten notetaking method.},
  file = {/home/cedarprince/Zotero/storage/35M92CSF/zettelkasten-how-one-german-scholar-was-so-freakishly-productive-997e4e0ca125.html},
  langid = {english},
  note = {Library Catalog: writingcooperative.com}
}
```

Instead, the frontmatter is getting rendered at the top like this:

![image](https://user-images.githubusercontent.com/29561456/76171794-935a9600-6165-11ea-8b54-50e8105291ed.png)

----

Here is the output of my gollum versions:

```sh
$ bundle exec bin/gollum --versions

Gollum 5.0.1b
Running on: x86_64-linux with Ruby version 2.6.5
Using:
gollum-grit_adapter 1.0.1
gollum-lib 5.0.a.4
rugged 0.28.4.1
gollum-rugged_adapter 0.4.4
With the following renderers:
kramdown 2.1.0
```

Any thoughts on what I am doing wrong? Am I using the YAML incorrectly? Thanks for the assistance!

**Comment by dometto** [comment]
@TheCedarPrince thanks for reporting this, that's a bug. Turns out github-markup is encountering the kramdown gem before it encounters pandoc-ruby, and hence uses the former. This is because you can't enforce the order of keys in ruby Hashes -- in previous rubies, it was encountering pandoc-ruby before kramdown by chance. Fix in https://github.com/gollum/gollum-lib/pull/356

(I'm assuming you have pandoc-ruby and citeproc-ruby installed, they unfortunately don't show up in the output of `--versions`.)

**Comment by dometto** [comment]
When the fix is merged (probably today/Monday), running `bundle install` should update `gollum-lib` and make everything work again 👍

**Comment by TheCedarPrince** [comment]
Hey @dometto,

Thanks for the speedy response! Looking forward to trying the fix when it is merged - I tried checking out your merge and applying the patch myself but am unfamiliar with Ruby architecture and did not get anywhere sadly.

If there is anything else you want me to check, just let me know. Also, I am running:

```rb
pandoc-ruby (2.0.2)
citeproc-ruby (1.1.12)
```

However, bundle cannot see them with the current GEMFILE:

```sh
$ bundle info pandoc-ruby

Could not find gem 'pandoc-ruby'.
```

When I modify the GEMFILE to this, it does not throw an error but can find the gem:

```rb
source 'https://rubygems.org'

if RUBY_PLATFORM == 'java'
  gem 'gollum-rjgit_adapter', :git => 'https://github.com/gollum/rjgit_adapter' # For development purposes
  gem 'warbler'
else
  gem 'gollum-rugged_adapter', :git => 'https://github.com/gollum/rugged_adapter.git' # For development purposes
end

gem 'gollum-lib', :git => 'https://github.com/gollum/gollum-lib.git', :branch => 'gollum-lib-5.x' # For development purposes
gem 'therubyrhino', :platforms => :jruby
gemspec
gem 'rake', '~> 11.2'
gem 'pandoc-ruby'
gem 'citeproc-ruby'
```

Is that necessary to do? Or should I just wait on the fix?

**Comment by dometto** [comment]
@TheCedarPrince yes you're right, you should add the gems to the Gemfile (as long as you're running gollum through `bundle exec` -- after we release 5.0, this won't be necessary anymore).

> When I modify the GEMFILE to this, it does not throw an error but can find the gem

Have you tried running `bundle install`?

**Comment by TheCedarPrince** [comment]
I tried running `bundle install` and even though I can see the gems per `bundle info`, I still have the same original error. What should I do now?

**Comment by dometto** [comment]
If you have the fix from https://github.com/gollum/gollum-lib/pull/356 that's puzzling! It's working well for me locally. Just to make sure: you've committed the `.bib` file to the repository?

**Comment by TheCedarPrince** [comment]
I did! I don't have the fix anymore - perhaps I should try a complete fresh install of gollum and try again?

Apologies for asking, but could you give me a step by step list of instructions on how to reproduce your results from a fresh install of gollum 5.x? I will attempt to reproduce it asap. Thanks @dometto !

**Comment by dometto** [comment]
The fix has been merged. Try to:
1. Freshly clone gollum
2. run `bundle install`
3. Profit

:)

If this problem persists, can you point me to an example repo?

**Comment by dometto** [comment]
@TheCedarPrince I was to quick in answering and had to update my answer above. Just wanted to let you know in case you were already trying it.

**Comment by TheCedarPrince** [comment]
It works - almost!

Right now, the citations do get rendered properly now. However, the YAML "table" thing still continues to show at the top of the page; any ideas why this is? Here is what it looks like:

![image](https://user-images.githubusercontent.com/29561456/76377981-9e0e5a00-6322-11ea-95d0-b7d5e9ad721e.png)

**Comment by dometto** [comment]
Gollum shows the YAML front matter for a page by default. You can disable this, though, as follows (from the wiki):

If you do not wish front matter to be displayed in this matter, there are two options:

If you do not wish front matter to be displayed in this matter, there are two options:

* disable globally by running gollum with the --no-display-metadata command line option
  *  or equivalently, by setting the wiki option :display_metadata in your config.rb to false
* disable on a per-page basis:
  * simply set display_metadata: false in your front matter!

**Comment by TheCedarPrince** [comment]
Perfect! Thanks for all the help @dometto ! :smile:

**Comment by TheCedarPrince** [comment]
@dometto , we need to re-open this fix as it is what is throwing the #1481 error. For some reason when my Gemfile looks like this:

```
source 'https://rubygems.org'

if RUBY_PLATFORM == 'java'
  gem 'gollum-rjgit_adapter', :git => 'https://github.com/gollum/rjgit_adapter' # For development purposes
  gem 'warbler'
else
  gem 'gollum-rugged_adapter', :git => 'https://github.com/gollum/rugged_adapter.git' # For development purposes
end

gem 'gollum-lib', :git => 'https://github.com/gollum/gollum-lib.git', :branch => 'gollum-lib-5.x' # For development purposes
gem 'therubyrhino', :platforms => :jruby
gemspec
gem "rake", ">= 12.3.3"
gem "pandoc-ruby"
gem 'bibtex-ruby'
gem 'citeproc-ruby'
gem 'csl-styles'
```

It throws that "untitled" error but would not do any citations. I don't know what to do here. Any thoughts?

**Comment by TheCedarPrince** [comment]
To provide additional context, here is my .csl file - maybe this is where the error is coming from?

```
<?xml version="1.0" encoding="utf-8"?>
<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only">
  <info>
    <title>IEEE</title>
    <id>http://www.zotero.org/styles/ieee</id>
    <link href="http://www.zotero.org/styles/ieee" rel="self"/>
    <link href="https://ieeeauthorcenter.ieee.org/wp-content/uploads/IEEE-Reference-Guide.pdf" rel="documentation"/>
    <link href="https://journals.ieeeauthorcenter.ieee.org/your-role-in-article-production/ieee-editorial-style-manual/" rel="documentation"/>
    <author>
      <name>Michael Berkowitz</name>
      <email>mberkowi@gmu.edu</email>
    </author>
    <contributor>
      <name>Julian Onions</name>
      <email>julian.onions@gmail.com</email>
    </contributor>
    <contributor>
      <name>Rintze Zelle</name>
      <uri>http://twitter.com/rintzezelle</uri>
    </contributor>
    <contributor>
      <name>Stephen Frank</name>
      <uri>http://www.zotero.org/sfrank</uri>
    </contributor>
    <contributor>
      <name>Sebastian Karcher</name>
    </contributor>
    <contributor>
      <name>Giuseppe Silano</name>
      <email>g.silano89@gmail.com</email>
      <uri>http://giuseppesilano.net</uri>
    </contributor>
    <category citation-format="numeric"/>
    <category field="engineering"/>
    <category field="generic-base"/>
    <summary>IEEE style as per the 2018 guidelines, V 11.12.2018.</summary>
    <updated>2019-12-20T09:14:19+00:00</updated>
    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
  </info>
  <locale xml:lang="en">
    <terms>
      <term name="chapter" form="short">ch.</term>
      <term name="presented at">presented at the</term>
      <term name="available at">available</term>
    </terms>
  </locale>
  <!-- Macros -->
  <macro name="status">
    <choose>
      <if variable="page issue volume" match="none">
        <text variable="status" text-case="capitalize-first" suffix="" font-weight="bold"/>
      </if>
    </choose>
  </macro>
  <macro name="edition">
    <choose>
      <if type="bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="any">
        <choose>
          <if is-numeric="edition">
            <group delimiter=" ">
              <number variable="edition" form="ordinal"/>
              <text term="edition" form="short"/>
            </group>
          </if>
          <else>
            <text variable="edition" text-case="capitalize-first" suffix="."/>
          </else>
        </choose>
      </if>
    </choose>
  </macro>
  <macro name="issued">
    <choose>
      <if type="article-journal report" match="any">
        <date variable="issued">
          <date-part name="month" form="short" suffix=" "/>
          <date-part name="year" form="long"/>
        </date>
      </if>
      <else-if type="bill book chapter graphic legal_case legislation motion_picture paper-conference song thesis" match="any">
        <date variable="issued">
          <date-part name="year" form="long"/>
        </date>
      </else-if>
      <else>
        <date variable="issued">
          <date-part name="day" form="numeric-leading-zeros" suffix="-"/>
          <date-part name="month" form="short" suffix="-" strip-periods="true"/>
          <date-part name="year" form="long"/>
        </date>
      </else>
    </choose>
  </macro>
  <macro name="author">
    <names variable="author">
      <name and="text" et-al-min="7" et-al-use-first="1" initialize-with=". "/>
      <label form="short" prefix=", " text-case="capitalize-first"/>
      <et-al font-style="italic"/>
      <substitute>
        <names variable="editor"/>
        <names variable="translator"/>
      </substitute>
    </names>
  </macro>
  <macro name="editor">
    <names variable="editor">
      <name initialize-with=". " delimiter=", " and="text"/>
      <label form="short" prefix=", " text-case="capitalize-first"/>
    </names>
  </macro>
  <macro name="locators">
    <group delimiter=", ">
      <text macro="edition"/>
      <group delimiter=" ">
        <text term="volume" form="short"/>
        <number variable="volume" form="numeric"/>
      </group>
      <group delimiter=" ">
        <number variable="number-of-volumes" form="numeric"/>
        <text term="volume" form="short" plural="true"/>
      </group>
      <group delimiter=" ">
        <text term="issue" form="short"/>
        <number variable="issue" form="numeric"/>
      </group>
    </group>
  </macro>
  <macro name="title">
    <choose>
      <if type="bill book graphic legal_case legislation motion_picture song" match="any">
        <text variable="title" font-style="italic"/>
      </if>
      <else>
        <text variable="title" quotes="true"/>
      </else>
    </choose>
  </macro>
  <macro name="publisher">
    <choose>
      <if type="bill book chapter graphic legal_case legislation motion_picture paper-conference song" match="any">
        <group delimiter=": ">
          <text variable="publisher-place"/>
          <text variable="publisher"/>
        </group>
      </if>
      <else>
        <group delimiter=", ">
          <text variable="publisher"/>
          <text variable="publisher-place"/>
        </group>
      </else>
    </choose>
  </macro>
  <macro name="event">
    <choose>
      <if type="paper-conference speech" match="any">
        <choose>
          <!-- Published Conference Paper -->
          <if variable="container-title">
            <group delimiter=", ">
              <group delimiter=" ">
                <text term="in"/>
                <text variable="container-title" font-style="italic"/>
              </group>
              <text variable="event-place"/>
            </group>
          </if>
          <!-- Unpublished Conference Paper -->
          <else>
            <group delimiter=", ">
              <group delimiter=" ">
                <text term="presented at"/>
                <text variable="event"/>
              </group>
              <text variable="event-place"/>
            </group>
          </else>
        </choose>
      </if>
    </choose>
  </macro>
  <macro name="access">
    <choose>
      <if type="webpage post post-weblog" match="any">
        <choose>
          <if variable="URL">
            <group delimiter=". ">
              <text term="online" prefix="[" suffix="]" text-case="capitalize-first"/>
              <group delimiter=": ">
                <text term="available at" text-case="capitalize-first"/>
                <text variable="URL"/>
              </group>
              <group prefix="[" suffix="]" delimiter=": ">
                <text term="accessed" text-case="capitalize-first"/>
                <date variable="accessed">
                  <date-part name="day" form="numeric-leading-zeros" suffix="-"/>
                  <date-part name="month" form="short" suffix="-" strip-periods="true"/>
                  <date-part name="year" form="long"/>
                </date>
              </group>
            </group>
          </if>
        </choose>
      </if>
      <else-if match="any" variable="DOI">
        <text variable="DOI" prefix="doi: "/>
      </else-if>
    </choose>
  </macro>
  <macro name="page">
    <group>
      <label variable="page" form="short" suffix=" "/>
      <text variable="page"/>
    </group>
  </macro>
  <macro name="citation-locator">
    <group delimiter=" ">
      <choose>
        <if locator="page">
          <label variable="locator" form="short"/>
        </if>
        <else>
          <label variable="locator" form="short" text-case="capitalize-first"/>
        </else>
      </choose>
      <text variable="locator"/>
    </group>
  </macro>
  <!-- Citation -->
  <citation collapse="citation-number">
    <sort>
      <key variable="citation-number"/>
    </sort>
    <layout delimiter=", ">
      <group prefix="[" suffix="]" delimiter=", ">
        <text variable="citation-number"/>
        <text macro="citation-locator"/>
      </group>
    </layout>
  </citation>
  <!-- Bibliography -->
  <bibliography entry-spacing="0" second-field-align="flush">
    <layout suffix=".">
      <!-- Citation Number -->
      <text variable="citation-number" prefix="[" suffix="]"/>
      <!-- Author(s) -->
      <text macro="author" suffix=", "/>
      <!-- Rest of Citation -->
      <choose>
        <!-- Specific Formats -->
        <if type="article-journal">
          <group delimiter=", ">
            <text macro="title"/>
            <text variable="container-title" font-style="italic" form="short"/>
            <text macro="locators"/>
            <text macro="page"/>
            <text macro="issued"/>
            <text macro="status"/>
            <text macro="access"/>
          </group>
        </if>
        <else-if type="paper-conference speech" match="any">
          <group delimiter=", ">
            <text macro="title"/>
            <text macro="event"/>
            <text macro="issued"/>
            <text macro="locators"/>
            <text macro="page"/>
            <text macro="status"/>
            <text macro="access"/>
          </group>
        </else-if>
        <else-if type="report">
          <group delimiter=", ">
            <text macro="title"/>
            <text macro="publisher"/>
            <group delimiter=" ">
              <text variable="genre"/>
              <text variable="number"/>
            </group>
            <text macro="issued"/>
          </group>
        </else-if>
        <else-if type="thesis">
          <group delimiter=", ">
            <text macro="title"/>
            <text variable="genre"/>
            <text macro="publisher"/>
            <text macro="issued"/>
          </group>
        </else-if>
        <else-if type="webpage post-weblog post" match="any">
          <group delimiter=", " suffix=". ">
            <text macro="title"/>
            <text variable="container-title" font-style="italic"/>
            <text macro="issued"/>
          </group>
          <text macro="access"/>
        </else-if>
        <else-if type="patent">
          <group delimiter=", ">
            <text macro="title"/>
            <text variable="number"/>
            <text macro="issued"/>
          </group>
        </else-if>
        <!-- Generic/Fallback Formats -->
        <else-if type="bill book graphic legal_case legislation motion_picture report song" match="any">
          <group delimiter=", " suffix=". ">
            <text macro="title"/>
            <text macro="locators"/>
          </group>
          <group delimiter=", ">
            <text macro="publisher"/>
            <text macro="issued"/>
            <text macro="page"/>
          </group>
        </else-if>
        <else-if type="article-magazine article-newspaper broadcast interview manuscript map patent personal_communication song speech thesis webpage" match="any">
          <group delimiter=", ">
            <text macro="title"/>
            <text variable="container-title" font-style="italic"/>
            <text macro="locators"/>
            <text macro="publisher"/>
            <text macro="page"/>
            <text macro="issued"/>
          </group>
        </else-if>
        <else-if type="chapter paper-conference" match="any">
          <group delimiter=", " suffix=", ">
            <text macro="title"/>
            <group delimiter=" ">
              <text term="in"/>
              <text variable="container-title" font-style="italic"/>
            </group>
            <text macro="locators"/>
          </group>
          <text macro="editor" suffix=" "/>
          <group delimiter=", ">
            <text macro="publisher"/>
            <text macro="issued"/>
            <text macro="page"/>
          </group>
        </else-if>
        <else>
          <group delimiter=", " suffix=". ">
            <text macro="title"/>
            <text variable="container-title" font-style="italic"/>
            <text macro="locators"/>
          </group>
          <group delimiter=", ">
            <text macro="publisher"/>
            <text macro="page"/>
            <text macro="issued"/>
            <text macro="access"/>
          </group>
        </else>
      </choose>
    </layout>
  </bibliography>
</style>

```

**Comment by dometto** [comment]
Happy to reopen this, but are you sure you are using the rugged adapter (and not the grit adapter) now? I was guessing that was what was causing #1481

**Comment by dometto** [comment]
@TheCedarPrince I see the following problems:

* When using pandoc-ruby, the h1 title is not respected
* When using pandoc-ruby **or not**, metadata title is not respected

However:

* I am not seeing the 'Untitled' problem. I'm still guessing this is because you were somehow using the grit adapter instead of the rugged adapter. (Use `--adapter rugged` to set the adapter explicitly)
* I **am** able to use pandoc-ruby's citation capabilities.

This is on the latest version of gollum:

```
Gollum 5.0.1b
Running on: x86_64-darwin18 with Ruby version 2.6.3
Using:
rugged 0.99.0
gollum-rugged_adapter 0.99.4
gollum-lib 5.0.a.4
Markdown rendering gem: pandoc-ruby
Other renderering gems:
none
```

We have been working on gollum, gollum-lib, and the adapters over the last couple of days, so it's possible you `git pull`ed an unstable version in at some point. Apologies for this. We're nearing the point where we can release 5.0 as a gem, which should hopefully obviate the need for any hacks.

**Comment by TheCedarPrince** [comment]
Oh dear. I don't know what is going on; I tried a fresh reinstall of everything. I deleted my old RVM Ruby environment, reinstalled it, checked out to the 5.x and did a fresh pull and now I cannot even execute `gollum --versions`

Here is the traceback:

```
bundler: failed to load command: gollum (gollum)
LoadError: cannot load such file -- rugged_adapter
  /home/src/Programs/RVM/gems/ruby-2.6.5/bundler/gems/gollum-lib-2b6c0c5a2133/lib/gollum-lib.rb:11:in `require'
  /home/src/Programs/RVM/gems/ruby-2.6.5/bundler/gems/gollum-lib-2b6c0c5a2133/lib/gollum-lib.rb:11:in `<top (required)>'
  gollum:173:in `require'
  gollum:173:in `block (2 levels) in <top (required)>'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1582:in `block in parse_in_order'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1568:in `catch'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1568:in `parse_in_order'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1562:in `order!'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1656:in `permute!'
  /home/src/Programs/RVM/rubies/ruby-2.6.5/lib/ruby/2.6.0/optparse.rb:1678:in `parse!'
  gollum:200:in `<top (required)>
```

Here are the contents of my Gemfile

```
source 'https://rubygems.org'

if RUBY_PLATFORM == 'java'
  gem 'warbler'
end

gem 'gollum-lib', :git => 'https://github.com/gollum/gollum-lib.git', :branch => 'gollum-lib-5.x' # For development purposes
gem 'therubyrhino', :platforms => :jruby
gemspec
gem "rake", ">= 12.3.3"
gem "pandoc-ruby"
gem "bibtex-ruby"
gem "citeproc-ruby"
gem "csl-styles"
```

Also here is the output of when I try to run `bundle exec bin/gollum [path-to-wiki] --adapter rugged`:

```
bundle exec bin/gollum ../ --adapter rugged
bundler: failed to load command: bin/gollum (bin/gollum)
LoadError: cannot load such file -- rugged_adapter
  /home/src/Programs/RVM/gems/ruby-2.6.5/bundler/gems/gollum-lib-2b6c0c5a2133/lib/gollum-lib.rb:11:in `require'
  /home/src/Programs/RVM/gems/ruby-2.6.5/bundler/gems/gollum-lib-2b6c0c5a2133/lib/gollum-lib.rb:11:in `<top (required)>'
  /home/src/VimWiki/gollum/lib/gollum/app.rb:5:in `require'
  /home/src/VimWiki/gollum/lib/gollum/app.rb:5:in `<top (required)>'
  bin/gollum:265:in `require'
  bin/gollum:265:in `<top (required)>'
```

I don't know what happened with the most recent updates but it seems I cannot even use gollum anymore. :cry: Any thoughts @dometto ?

**Comment by bartkamphorst** [comment]
@dometto After pulling in the latest changes from gollum and gollum-lib, I'm seeing the exact same LoadError as @TheCedarPrince is reporting.

**Comment by bartkamphorst** [comment]
The LoadError stems from the fact that `gem 'gollum-rugged_adapter'` is missing from the Gemfile. When I set
```ruby
if RUBY_PLATFORM == 'java'
  gem 'warbler'
  gem 'gollum-rjgit_adapter'
else
  gem 'gollum-rugged_adapter'
end
```
then gollum starts correctly.

**Comment by dometto** [comment]
Are you sure there isn't something else going wrong here, @bartkamphorst? The adapters are now defined [here](https://github.com/gollum/gollum-lib/blob/gollum-lib-5.x/gollum-lib.gemspec), on gollum-lib's gollum-lib-5.x branch. So I think it shouldn't be necessary to add the adapters to gollum's own Gemfile.I am in fact able to run gollum on the 5.x branch without any modifications to the Gemfile, and the travis tests also pass. Perhaps you are somehow not using the correct version of gollum-lib?

**Comment by dometto** [comment]
@TheCedarPrince I'm having no trouble running Gollum with your Gemfile, a fresh 5.x branch, and a fresh rvm environment:

```
Gollum 5.0.1b
Running on: x86_64-darwin17 with Ruby version 2.5.3
Using:
rugged 0.99.0
gollum-rugged_adapter 0.99.4
gollum-lib 5.0.a.4
Markdown rendering gem: pandoc-ruby
Other renderering gems:
none
```

This is making me suspect you still have some remnants from your old set up hidden somewhere. Perhaps an old `Gemfile.lock` was preventing you from installing the newest gollum-lib? Try removing `Gemfile.lock` and then running `bundle install` again?

**Comment by bartkamphorst** [comment]
> Perhaps an old Gemfile.lock was preventing you from installing the newest gollum-lib?

This was my issue. To resolve, I took the following steps:
* bundle exec gem uninstall gollum-grit_adapter
* bundle update

(This reported "Using gollum-lib 5.0.a.4 from https://github.com/gollum/gollum-lib.git (at gollum-lib-5.x@603f08e)")

Then bundle exec gollum --versions showed this:
```
$ bundle exec gollum --adapter rugged --versions
Gollum 5.0.1b
Running on: [...] with Ruby version 2.6.3
Using:
rugged 0.99.0
gollum-rugged_adapter 0.99.4
gollum-lib 5.0.a.4
Markdown rendering gem: kramdown
Other renderering gems:
none
```

Then I was able to start gollum with `bundle exec gollum ../gollum-wiki`. 

(Note that my Gemfile did not reference the rugged adapter in this setup.)

**Comment by dometto** [comment]
👍 I'm going to assume that solves the issue then, but feel free to post here if you require more assistance, @TheCedarPrince

I've opened a separate issue for the difficulties with h1 title: #1484

**Comment by TheCedarPrince** [comment]
That fixed it! Thanks @bartkamphorst and @dometto !
